packages/shared/src/reparent-children.ts

```diff
@@ -33,19 +33,24 @@ function restoreChildren<T extends Element>(
 export const reparentChildren = <T extends Element>(
     srcElements: T[],
     newParent: Element,
-    prepareCallback?: (el: T) => ((el: T) => void) | void
+    prepareCallback?: ((el: T) => ((el: T) => void) | void) | null,
+    position?: InsertPosition
 ): (() => Element[]) => {
     const placeholderItems: Comment[] = [];
     const cleanupCallbacks: ((el: T) => void)[] = [];
 
+    // default is to append
+    position = position || 'beforeend';
+
+    if (position === 'afterbegin' || position === 'afterend') {
+        srcElements.reverse();
```

Comment:
At what scale do you think `reverse()` would negatively effect performance here as opposed to configuring the loop to walk through the array in a different direction, instead? Nothing leverages this internally at this point, but it might be worth getting a stress test benchmark together. The `<sp-split-button>` benchmark is the closest to what that might look like: https://github.com/adobe/spectrum-web-components/blob/main/packages/split-button/test/benchmark/basic-test.ts

Reply:
`cloneNode()` definitely looks good to me! We'd not tracked that as an option, but we do do a growing amount of reparenting, so continuing to trim down the time in the method is going to be a win for everyone.